modules/nf-httpfs/src/main/nextflow/file/http/LatchFileSystemProvider.groovy

```diff
@@ -57,37 +57,40 @@ class LatchFileSystemProvider extends XFileSystemProvider {
         if (fileSystems.containsKey(domain))
             throw new FileSystemAlreadyExistsException()
 
-        LatchFileSystem fs = new LatchFileSystem(this, domain)
-        fileSystems[domain] = fs
```

Comment:
i removed this and moved it to the getFileSystem function because that is consistent with the XFileSystemProvider. not sure if this is wrong tho. seems like newFileSystem is only used in one other place (FileHelper.groovy)

Reply:
probably fine